tests/Google/Ads/GoogleAds/Util/FieldMasksTest.php

```diff
@@ -135,15 +137,32 @@ public function testGetFieldValue()
                     ->getIterator()
             )
         );
+        // A value of a repeated field type of a message can be obtained.
```

Comment:
Could you also add a unit test that covers when the field value  path traverses an array field?

Typically, the field value paths `url_custom_parameters.key` and `url_custom_parameters[0].key` should raise the `UnexpectedValueException` exception.

Reply:
`url_custom_parameters[0].key` should raise the exception because `xxx[0]` is originally not the form that this method can handle. I think we need to first confirm if this form is ever returned by the API.

This fix aims to at least able to handle the repeated field of messages. I think we need to first define if `xxx[0]` is a valid field mask that can be returned by the ChangeEvent resource.
I tried to reproduce #512, but unfortunately I cannot get the real use case of it too, so I'm not sure that the user gets a repeated field of null (messages) or if the user really gets a repeated field of non-empty messages (with a path like `url_custom_parameters[0].key`, as you mentioned.

@AnashOommen WDYT? Looks like you're the one who worked on this example last time?

Reply:
Syncing with Anash offline. Turns out Anash believes the repeated fields shouldn't be returned as a field mask path by ChangeEvent. So I'm gonna roll the change about it back.
We can get back to this one we have more evidence that the field mask path can be repeated fields.